uobtheatre/bookings/mutations.py

```diff
@@ -424,6 +429,49 @@ def resolve_mutation(cls, _, info, booking_reference, tickets, performance):
         return UnCheckInBooking(booking=booking, performance=performance)
 
 
+class CreateBookingTransfer(AuthRequiredMixin, SafeMutation):
+    """Mutation to transfer a booking to a difference performance. This creates
+    an inprogress booking (transfer) which is linked to the orignal booking via
+    transferred_to.
+
+    Args:
+        booking_id (str): The gloabl id of the Booking to transfer.
+        performance_id (str): The gloabl id of the performance which the
+            Booking should be transfered to.
+
+    Returns:
+        booking (BookingNode): The Booking for the new performance.
+
+    Raises:
+        GQLException: If the Payment was unsucessful.
+    """
+
+    booking = graphene.Field(BookingNode)
+
+    class Arguments:
+        booking_id = IdInputField(required=True)
+        performance_id = IdInputField(required=True)
+
+    @classmethod
+    def resolve_mutation(  # pylint: disable=too-many-branches
+        cls,
+        _,
+        info,
+        booking_id,
+        performance_id,
+    ):
+        booking = Booking.objects.get(pk=booking_id, user=info.context.user.id)
+        performance = Performance.objects.get(pk=performance_id)
+
+        if not BookForPerformance.user_has_for(info.context.user, performance):
```

Comment:
We also need to make sure they have permission to do this transfer? I.e. they own the booking (or are box office but this requires some extra work because we need to set created by user of the new booking)

Reply:
@atoff this has your name all over it